python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets.py

```diff
@@ -673,3 +679,158 @@ def foo():
 
     the_asset = AssetsDefinition.from_op(op_def=foo.configured({"bar": "abc"}, name="foo2"))
     assert the_asset.keys_by_output_name["result"].path == ["foo2"]
+
+
+def get_step_keys_from_run(instance):
+    engine_events = list(
+        instance.get_event_records(EventRecordsFilter(DagsterEventType.ENGINE_EVENT))
+    )
+    metadata_entries = engine_events[
+        0
+    ].event_log_entry.dagster_event.engine_event_data.metadata_entries
+    step_metadata = next(
+        iter([metadata for metadata in metadata_entries if metadata.label == "step_keys"])
+    )
+    return ast.literal_eval(step_metadata.value.value)
+
+
+def test_graph_backed_asset_subset():
+    @op()
+    def foo():
+        return 1
+
+    @op
+    def bar(foo):
+        return foo
+
+    @graph(out={"one": GraphOut(), "two": GraphOut()})
+    def my_graph():
+        one = foo()
+        return bar.alias("bar_1")(one), bar.alias("bar_2")(one)
+
+    asset_job = define_asset_job("yay").resolve(
+        [
+            AssetsDefinition.from_graph(my_graph, can_subset=True),
+        ],
+        [],
+    )
+
+    with instance_for_test() as instance:
+        asset_job.execute_in_process(instance=instance, asset_selection=[AssetKey("one")])
+        materialization_planned = list(
+            instance.get_event_records(
+                EventRecordsFilter(DagsterEventType.ASSET_MATERIALIZATION_PLANNED)
+            )
+        )
+        assert len(materialization_planned) == 1
+        step_keys = get_step_keys_from_run(instance)
+        assert set(step_keys) == set(["my_graph.foo", "my_graph.bar_1"])
+
+
+def test_graph_backed_asset_partial_output_selection():
+    @op(out={"a": Out(), "b": Out()})
+    def foo():
+        return 1, 2
+
+    @graph(out={"one": GraphOut(), "two": GraphOut()})
+    def graph_asset():
+        one, two = foo()
+        return one, two
+
+    asset_job = define_asset_job("yay").resolve(
+        [
+            AssetsDefinition.from_graph(graph_asset, can_subset=True),
+        ],
+        [],
+    )
+
+    with instance_for_test() as instance:
+        asset_job.execute_in_process(instance=instance, asset_selection=[AssetKey("one")])
+        materialization_planned = list(
+            instance.get_event_records(
+                EventRecordsFilter(DagsterEventType.ASSET_MATERIALIZATION_PLANNED)
+            )
+        )
+        assert len(materialization_planned) == 1
```

Comment:
bit of a nitpick, but in all of these tests, do you mind adding a check for the actual number of asset materialization events (in addition to the number of planned events)?

This one is a bit of a special case, as it will emit more materializations than planned, but the others should match up I believe.